cmd/training-operator.v1/main.go

```diff
@@ -187,3 +193,17 @@ func setupControllers(mgr ctrl.Manager, enabledSchemes controllerv1.EnabledSchem
 		}
 	}
 }
+
+func ValidateCRD(mgr ctrl.Manager, gvk schema.GroupVersionKind) {
+	_, err := mgr.GetRESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		if meta.IsNoMatchError(err) {
+			setupLog.Error(err, "crd might be missing, please install crd",
+				"crd group", gvk.Group, "crd version", gvk.Version, "crd kind", gvk.Kind)
```

Comment:
```suggestion
				"group", gvk.Group, "version", gvk.Version, "kind", gvk.Kind)
```

The "crd" looks redundant. And we should output logs in a JSON-like format for the structured logging.

Reply:
```
setupLog.Error(err, "crd might be missing, please install crd", "gvk", gvk.String())

ERROR   setup   crd might be missing, please install crd        {"gvk": "scheduling.volcano.sh/v1beta1, Kind=PodGroup", "error": "failed to get API group resources: unable to retrieve the complete list of server APIs: scheduling.volcano.sh/v1beta1: the server could not find the requested resource"}
```
@tenzen-y  Is it okay to modify it like this?

Reply:
I meant the below:

```
ERROR   setup   crd might be missing, please install crd        {"group": "scheduling.volcano.sh", "version": "v1beta1", "kind", "PodGroup", "error": "failed to get API group resources: unable to retrieve the complete list of server APIs: scheduling.volcano.sh/v1beta1: the server could not find the requested resource"}
```

Reply:
Or 

```go
..., "apiVersion": gvk.GroupVersion().String(), "kind": "gvk.Kind")
```

```shell
{"apiVersion": "scheduling.volcano.sh/v1beta1", "kind": "PodGroup", "error": ""...
```

Reply:
Done, thanks